quixstreams/app.py

```diff
@@ -710,6 +741,31 @@ def clear_state(self):
         """
         self._state_manager.clear_stores()
 
+    def source(self, source: BaseSource, topic: Optional[Topic] = None) -> Topic:
+        """
+        Add a source to the application.
+
+        See :class:`quixstreams.sources.base.BaseSource` for more details.
+
+        :param source: a :class:`quixstreams.sources.BaseSource` instance
+        :param topic: a :class:`quixstreams.models.Topic` instance to be used as the source producer topic
+            Default: `None`
+        """
+        if not topic:
+            topic = source.default_topic(self._topic_manager)
+
+        producer = RowProducer(
+            broker_address=self._broker_address,
+            extra_config=self._producer_extra_config,
+            flush_timeout=self._consumer_extra_config.get(
+                "max.poll.interval.ms", _default_max_poll_interval_ms
+            )
+            / 1000,  # convert to seconds
+            transactional=self._uses_exactly_once,
```

Comment:
In this case, I think we want to use ALO regardless (or at the very least, no use the app setting), as you can't really promise EOS with Sourcing except for maybe some special circumstances, no?

Also just better to be safe as there are other configurations in place when this is set to EOS.

Reply:
To that end, might even want to warn users somehow and not give them a false impression of the processing guarantees provided with a source.

Reply:
Yes I think we can start by having it set to ALO and maybe later some sources can leverage EOS